src/js/media-controller.js

```diff
@@ -62,57 +61,57 @@ class MediaController extends MediaContainer {
 
   #hotKeys = new AttributeTokenList(this, Attributes.HOTKEYS);
   #fullscreenElement;
+  #mediaState = {};
 
   constructor() {
     super();
 
-    // Update volume support ASAP
-    if (MediaUIStates.MEDIA_VOLUME_UNAVAILABLE.get(this) === undefined) {
-      // NOTE: In order to propagate ASAP, we currently need to ensure that
-      // the volume support promise has resolved. Given the async nature of
-      // some of these environment state values, we may want to model this
-      // a bit better (CJP).
-      volumeSupportPromise.then(() => {
-        this.propagateMediaState(
-          MediaUIAttributes.MEDIA_VOLUME_UNAVAILABLE,
-          MediaUIStates.MEDIA_VOLUME_UNAVAILABLE.get(this)
-        );
-      });
-    }
-
     // Track externally associated control elements
     this.mediaStateReceivers = [];
     this.associatedElementSubscriptions = new Map();
     this.associateElement(this);
 
-    // Apply ui event listeners to self
-    // Should apply to any UI container, not just controller
-    Object.keys(MediaUIRequestHandlers).forEach((key) => {
-      const handlerName = `_handle${constToCamel(key, true)}`;
-
-      // TODO: Move to map, not obj root property
-      this[handlerName] = (e) => {
-        // Stop media UI events from continuing to bubble
-        e.stopPropagation();
-
-        if (!this.media) {
-          console.warn('MediaController: No media available.');
-          return;
-        }
-
-        MediaUIRequestHandlers[key](this.media, e, this);
-      };
-      this.addEventListener(MediaUIEvents[key], this[handlerName]);
+    this._mediaStore = createMediaStore({
+      media: this.media,
+      // @ts-ignore
+      rootNode: this.getRootNode(),
+      fullscreenElement: this.fullscreenElement,
+      // NOTE: Names are hard. What should these be called (both the "bucket" currently called config and the props)
+      config: {
+        subtitles: this.hasAttribute('defaultsubtitles'),
```

Comment:
The name "subtitles" here is pretty vague. I think in general it'd be good to match these option names with the attr name, unless there's a good reason not to.

Reply:
awesome yeah after getting further in the impl I was actually thinking the same thing.